.travis.yml

```diff
@@ -1,13 +1,45 @@
 sudo: false
 language: node_js
-script:
-  - npm run build
-  - npm run test:ci
 node_js:
+  - 6
   - 8
   - 10
   - 12
 
+before_install:
+  - npm config set depth 0
+before_script: >
+  node_version=$(node -v);
+  if [ ${node_version:3:1} = "." ]; then
+    echo "Node 10+"
+    npm install --no-save "eslint@6"
+    npm run build
+    npm run test:ci
+    else
+      if [ ${node_version:1:1} -ge 8 ]; then
+        echo "Node 8+"
+        npm install --no-save "eslint@5"
+        npm run build
+        npm run test:ci
+      else
+        if [ ${node_version:1:1} -ge 6 ]; then
+          echo "Node 6+"
+          npm install --no-save "eslint@5" mocha@6.2.2 nyc@14.1.1
+          npm run test:ci
```

Comment:
How are we running the tests without building (`npm run build`) on this platform?

Reply:
Oh I see now. By not running the build step, we're testing against the old parser and will fail when we go to add a new feature.

Reply:
I don't recall if the build routine works or not (I can try for parity if you like), but it is not needed because the tests are able to run using `@babel/register` testing against the source instead of `dist` files.

(I only needed to switch to testing the `dist` files in the branch where I manually confirmed the old Node builds were still working. But that branch had to undo ESM in the test files and fixtures, avoid other ES6 features, as well as revert to the vulnerable Mocha 3.)

Reply:
I don't understand. Why would we be able to run "against the source" using `@babel/register` but not compile? Isn't `@babel/register` just compiling on the fly and running that?

Reply:
On the much older Node versions, `@babel/register` doesn't work, and `babel-register`  doesn't seem to be able to understand the import statements as @babel/register does.

As far as Node 6 that is failing on Travis, yes, `@babel/register` is compiling on the fly so the babel component should be doable, but it appears that the current `@rollup/pluginutils` (used by the current versions of `@rollup/plugin-commonjs` and `@rollup/plugin-node-resolve` needed in the Rollup config file) is using object spread properties which weren't supported in Node 6. Note that it wouldn't be enough to go back just to an older version here either, as these were previously not in the `@rollup` scope but were their own packages. But if we go too far back in our toolchain, then certain other features might not have been supported then.

Is it really critical that the build scripts are confirmed to work? These aren't available to work in the npm releases anyways (since the rollup config is not in `package.json` `files` and Rollup, etc. are devDeps).

This should really only be needed for those compiling from source on the repo who would presumably have access to more up to date machines. Seems it may be some unnecessary work for little benefit (not to mention adding to the travis build time).

But I can try to see if we can get it to work for Node 6 if it is important.